src/elements.h

```diff
@@ -567,6 +573,17 @@ namespace libsemigroups {
       return new T(vector);
     }
 
+    //! Returns boolean representing whether or not a partial transformation is
+    //! the identity.
+    bool is_identity() {
```

Comment:
It would be better if this was a deleted method for the `Element` class and there was a method for every type of `Element`. Again this does not seems related to the introduction of the `Permutation` class, so you might consider removing this completely from here, and making a separate PR with this complete change.